File: apps/dotnet/Simex/Simex/Models/Country.cs
using System;
using System.Collections.Generic;

namespace Simex.Models;

public partial class Country
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public virtual ICollection<City> Cities { get; set; } = new List<City>();

    public virtual ICollection<Region> Regions { get; set; } = new List<Region>();
}

File: apps/web/backend/app/Models/Country.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasManyThrough; 

class Country extends Model
{
    protected $table = 'COUNTRIES';
    protected $primaryKey = 'ID';
    public $timestamps = false;

    protected $fillable = [
        'NAME',
    ];

    public function regions(): HasMany
    {
        return $this->hasMany(Region::class, 'COUNTRY_ID', 'ID');
    }

    
     
    public function cities(): HasManyThrough 
    {
        return $this->hasManyThrough(City::class, Region::class, 'COUNTRY_ID', 'REGION_ID', 'ID', 'ID');
    }
}